barman/config.py

```diff
@@ -876,6 +878,9 @@ def _parse_global_config(self):
         self.barman_lock_directory = (
             self.get("barman", "barman_lock_directory") or self.barman_home
         )
+        self.lock_directory_cleanup = self.get(
```

Comment:
`parse_boolean` is only called on the default and not on the value of `self.get("barman", "lock_directory_cleanup")` which means any value entered in the config is going to be a string rather than a boolean so a config of `lock_directory_cleanup = false` will not currently cause the cleanup to be skipped.

Reply:
right, should be applied to both `lock_directory_cleanup` and `DEFAULT_CLEANUP`